src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ArrayTrailingCommaCheck.java

```diff
@@ -164,8 +205,9 @@ public void visitToken(DetailAST arrayInit) {
         final DetailAST previousSibling = rcurly.getPreviousSibling();
 
         if (arrayInit.getChildCount() != 1
-                && !TokenUtil.areOnSameLine(rcurly, previousSibling)
-                && !TokenUtil.areOnSameLine(arrayInit, previousSibling)
+                && (alwaysDemandTrailingComma || !TokenUtil.areOnSameLine(rcurly, previousSibling))
+                && (alwaysDemandTrailingComma
+                    || !TokenUtil.areOnSameLine(arrayInit, previousSibling))
```

Comment:
I wonder if it would be easier/harder to read if pulling out `alwaysDemandTrailingComma`, e.g.
```java
&& (alwaysDemandTrailingComma ||
    (!TokenUtil.areOnSameLine(rcurly, previousSibling) &&
     !TokenUtil.areOnSameLine(arrayInit, previousSibling)))
```

Reply:
Done